### PR TITLE
Closing wine bug 46237 by not setting the cursor position in the wrong moment.

### PR DESCRIPTION
**Pull request by TheChatty**

### CODE REVIEW

**Comment by TheChatty** [comment]
At first I thought of a simple race condition which it is not, as it turns out. Execution flawlessly passes `SetCursorPosition` and causes Far to crash roughly one second later - **but only** if `SetCursorPosition` gets executed.

Since before 21bef08 `InitConsole` didn't contain `SetCursorPosition` and as Far was working correctly after removing it I wasn't "_randomly_" removing code.

Calling `ResetPosition` instead (like before) still also works.

Analyzing `ScreenBuf::Scroll` I found that Far crashes randomly [here (A)](https://github.com/FarGroup/FarManager/blob/3ab95f5184a031feebea6624d0e525a6c100bc17/far/scrbuf.cpp#L599) or [here (B)](https://github.com/FarGroup/FarManager/blob/3ab95f5184a031feebea6624d0e525a6c100bc17/far/scrbuf.cpp#L603).
Removing (A) only, I never saw it passing (B) - but no crash, always hang.
Removing (B) only, Far starts should it pass (A) by chance.
Removing (A)+(B), Far starts as usual.

File: far/interf.cpp [hunk 1]
```diff
@@ -309,9 +309,6 @@ void InitConsole()
 
 		if(Global->Opt->WindowMode)
 		{
-			COORD Size;
```
[the remark this answers is not in the document]
Comment (TheChatty) [reply far/interf.cpp]: Please consider below statement.

**Comment by alabuzhev** [comment]
Ok, by "randomly" (no offence) I mean that trying to make Wine happy doesn't make much sense because it's a hide-and-seek game.
Moving the cursor to the bottom line is a perfectly legal thing. Far moves it all the time, so if it crashes here it could as well crash somewhere else.

Does it happen if you call ResetPosition **before** SetCursorPosition?

File: far/interf.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (alabuzhev) [reply far/interf.cpp]: Up.
This code has been slightly reorganised due do the recent changes in Windows 10 which, amusingly, also introduced crashes related to the cursor position.

@TheChatty, does the latest master still crash in Wine?

[the remark this answers is not in the document]
Comment (TheChatty) [reply far/interf.cpp]: @alabuzhev: Unfortunately I can no longer evaluate because I no longer own the affected MacBook.

**Pull request closed without merging**